### ISSUE TITLE
Importation of a FBX, which has bones with the same name, causes a crash

### ISSUE THREAD

**Issue by johannnovak** [assimp/assimp#5085]
Hello,

I encountered an issue when importing a mesh which possesses two bones with the same name.
It crashes on the destruction of the Assimp::Importer after an importation has occured within its lifespan.
The destruction implies the destruction of a aiScene which implies the destruction of the aiMesh array, and then their aiBones.

For the destruction of a aiMesh, this code is called (for the bones):
```cpp
        // DO NOT REMOVE THIS ADDITIONAL CHECK
        if (mNumBones && mBones) {
            for (unsigned int a = 0; a < mNumBones; a++) {
                if (mBones[a]) {
                    delete mBones[a];
                }
            }
            delete[] mBones;
            mBones = nullptr;
        }
```
        
In my case, the bone at index 57 and 58 points to the same aiBone. So when the 57th is deleted, the 58th points on deleted data and the delete is called on it causing a crash.

I think this comes from the FBXConverter which stores the bones inside a map :

```// Deformer name is not the same as a bone name - it does contain the bone name though :)
    // Deformer names in FBX are always unique in an FBX file.
    std::map<const std::string, aiBone *> bone_map;
```

This map is filled inside the method "ConvertCluster".

My version of assimp is 5.2.3.

**Comment by kimkulling** [comment]
Yes, this is an issue. We need to remember which bone was deleted already or we need a different data structure to store the referenced bones.

**Comment by kimkulling** [comment]
Or just looking of the name is already in. I will try a PR to help you out.

**Comment by kimkulling** [comment]
Just checked, and the name will be checked before it will be inserted. The description of the infected code seems to be not completely correct.

**Comment by tellypresence** [comment]
@johannnovak Please attach problematic 3D model file to issue for investigation

**Comment by tellypresence** [comment]
Closing due to no response to followup
